### PR TITLE
Add docstrings to make_release.py (#8)

### LINKED ISSUE #8: make_release scripts needs docstrings
https://github.com/DIRACGrid/DIRACOS2/blob/5a5483b0eb0a7b2a34a19a72c46610f29012db90/scripts/make_release.py#L1

For #7 

For maintainability, the script needs a number of docstrings to explain what it does and what the individual functions do. They can be brief, but at the moment there are none.

https://github.com/DIRACGrid/DIRACOS2#making-a-release could refer to the make_release script

### PR DESCRIPTION
**Pull request by chrisburr**
Closes #8

### CODE REVIEW

**Comment by chrisburr** [comment]
Ping @andresailer

File: scripts/make_release.py [hunk 1]
```diff
@@ -15,43 +14,31 @@
...
-        ]
-
-    return str(version), "".join(next_version)
-
-
 def main(run_id=None, requested_version=None, workflow_fn="build-and-test.yml"):
+    """Make a new release of DIRACOS2 based on a GitHub Actions CI run
+
+    Releases are made by:
+    1. Download the artefact from a GitHub Actions CI run
```
Comment (andresailer) [inline scripts/make_release.py:21]: newline before enumerated list

Comment (andresailer) [inline scripts/make_release.py:21]:
```suggestion
    1. Download the artifact from a GitHub Actions CI run
```
Cf. `get_artifacts_zip` below. I know the "e" is the British spelling, but consistency.

File: scripts/make_release.py [hunk 2]
```diff
@@ -15,43 +14,31 @@
...
+    2. Editing it in place to change the version number
+    3. Creating a pre-release in GitHub
+    4. Uploading release artefacts to the new pre-release
```
Comment (andresailer) [inline scripts/make_release.py:24]:
```suggestion
    4. Uploading release artifacts to the new pre-release
```

File: scripts/make_release.py [hunk 3]
```diff
@@ -87,7 +74,91 @@ def main(run_id=None, requested_version=None, workflow_fn="build-and-test.yml"):
...
+        next_version = [
+            ".".join(map(str, version.release[:-1] + (version.release[-1] + 1,))),
+            "a1"
+        ]
+
+    return str(version), "".join(next_version)
+
+
 def get_installer_artifacts(run_id):
+    """Download the artefacts from a GitHub Actions run
```
Comment (andresailer) [inline scripts/make_release.py:144]:
```suggestion
    """Download the artifacts from a GitHub Actions run
```

File: scripts/make_release.py [hunk 4]
```diff
@@ -87,7 +74,91 @@ def main(run_id=None, requested_version=None, workflow_fn="build-and-test.yml"):
...
+
+    Parameters
+    ----------
+    run_id: int
+        The GitHub Actions run ID to download artifacts from
+
+    Returns
+    -------
+    A tuple of the:
+     * Git revision used for the given Run ID
```
Comment (andresailer) [inline scripts/make_release.py:154]: Just for the list, newline, no additional indent.

File: scripts/make_release.py [hunk 5]
```diff
@@ -111,6 +182,25 @@ def get_installer_artifacts(run_id):
 
 
 def make_release(installer, environment_yaml, version, commit_hash):
+    """Create a new GitHub release using the given data
+
+    This function always makes a pre-release first to ensure the "latest" release never corresponds
+    to one without artefacts uploaded. If the new version number is not a pre-release, as
+    determined by PEP-440, it is promoted to at full release after he uploads have completed
```
Comment (andresailer) [inline scripts/make_release.py:189]:
```suggestion
    determined by PEP-440, it is promoted to at full release after the uploads have completed
```

File: scripts/make_release.py [hunk 6]
```diff
@@ -111,6 +182,25 @@ def get_installer_artifacts(run_id):
 
 
 def make_release(installer, environment_yaml, version, commit_hash):
+    """Create a new GitHub release using the given data
+
+    This function always makes a pre-release first to ensure the "latest" release never corresponds
+    to one without artefacts uploaded. If the new version number is not a pre-release, as
```
Comment (andresailer) [inline scripts/make_release.py:188]:
```suggestion
    to one without artifacts uploaded. If the new version number is not a pre-release, as
```

File: scripts/make_release.py [hunk 7]
```diff
@@ -15,43 +14,31 @@
...
+    3. Creating a pre-release in GitHub
+    4. Uploading release artefacts to the new pre-release
+    5. If the version number is not a pre-release: converting the pre-release to a full release
+       causing the "latest" release to update
+    6. Editing the current "main" branch to be one version higher than the current latest release
+
+    Authentication is performed using the GitHub API token which is automatically made available
+    to all GitHub Actions jobs.
+
+    Parameters
```
Comment (andresailer) [inline scripts/make_release.py:32]: Can we just use the normal sphinx documentation syntax for parameters and returns?

**Review by andresailer — COMMENTED** [review]
Sorry for missing this so long:
To fully solve #8, please refer to this script from https://github.com/DIRACGrid/DIRACOS2#making-a-release

File: scripts/make_release.py (hunk 7 above)
Comment (chrisburr) [reply scripts/make_release.py:32]: Definitely, can you point me to an example as I can never remember which style is which

Comment (andresailer) [reply scripts/make_release.py:33]: Thanks!
Like here (including newline before), `:param theType name:` or `:param name:`, `:type name: theType or thisType`
https://github.com/DIRACGrid/DIRAC/blob/841e68f1880fa72c74d27a4876ba2ee2c4f72a72/TransformationSystem/Utilities/ReplicationTransformation.py#L23-L37

**Comment by andresailer** [comment]
Last thing still please refer to this script from https://github.com/DIRACGrid/DIRACOS2#making-a-release

**Comment by chrisburr** [comment]
Sorry I missed that remark, it should be done now.

**Review by andresailer — APPROVED** [review]
Many thanks!

**Pull request merged**